encode_utils/connection.py

```diff
@@ -1227,6 +1227,131 @@ def remove_props(self, rec_id, props=[]):
+        This is useful because some schema dependencies requires property
+        removal and property patch (including adding new properties) happening
+        at the same time. Please note that after the record is retrieved from
+        the portal, ``props`` will be removed before the ``patch`` is applied.
+
+        Args:
+            props: `list`. The properties to remove from the record.
+            patch: `dict`. containing the attribute key and value pairs to
+                patch. Must contain the key ``self.ENCID_KEY`` in order to
+                indicate which record to PATCH.
+            raise_403: `bool`. `True` means to raise a
+                ``requests.exceptions.HTTPError`` if a 403 status (forbidden)
+                is returned. If set to `False` and there still is a 403 return
+                status, then the object you were trying to PATCH will be
+                fetched from the Portal in JSON format as this function's
+                return value.
+            extend_array_values: `bool`. Only affects keys with array values.
+                `True` (default) means to extend the corresponding value on the
+                Portal with what's specified in the payload. `False` means to
+                replace the value on the Portal with what's in the payload.
+
+        Returns:
+            `dict`: The JSON response from the PUT operation, or an empty dict
+                if the record doesn't exist on the Portal. Will also be an
+                empty dict if the dry-run feature is enabled.
+
+        Raises:
+            requests.exceptions.HTTPError: if the return status is not ok
+                (excluding a 403 status if 'raise_403' is False).
+        """
+        # Make sure we have a payload that can be converted to valid JSON, and
+        # tuples become arrays, ...
+        patch = json.loads(json.dumps(patch))
+        self.debug_logger.debug("\nIN remove_and_patch()")
+        encode_id = patch[self.ENCID_KEY]
+        rec_json = self.get(rec_ids=encode_id, frame="object", ignore404=True)
+        if not rec_json:  # Ensure that the record exists on the Portal:
+            return {}
+        profile = eup.Profile(rec_json["@id"])
+        payload = self.get(rec_ids=encode_id, frame="edit", ignore404=False)
+        for prop in props:
+            if profile.is_prop_required(prop):
+                raise Exception("Can't remove required property")
+            elif profile.is_prop_not_submittable(prop):
+                raise Exception("Can't remove non-submittable property.")
+            elif profile.is_prop_read_only(prop):
+                raise Exception("Can't remove read-only property.")
+            else:
+                # Then it is safe to remove this property.
+                payload.pop(prop, None)
+        for key in patch:
+            if (
+                extend_array_values
+                and isinstance(patch[key], list)
+                and key in payload
+            ):
```

Comment:
This would be better written using `all()`:
```
            if all(
                extend_array_values,
                isinstance(patch[key], list),
                key in payload
            ):
```

Reply:
`all()` takes in only one iterable argument. I put them in a list.

Reply:
Oh yeah you're right, that always gets me.